### DISCUSSION TITLE
compile/lab2

### DISCUSSION THREAD

**Discussion by giscus** [jielahou/blog-comments#2] [Announcements]
# compile/lab2

实验二要求依据我们在实验一建立起来的语法分析树，进行语义分析。 过程概览 词法、语法分析，先得到一棵语法分析树； 对语法分析树进行遍历，进行语义分析，（如果没有错误）得到语义分析表； 上课讲的声明语句(即填表)例子，是将填表过程(SDT)融合在语法分析过程(Bison)中的。 而实验手册的意思是先有树，再单独分析；可是，分析的过程是需要用到SDT的哇！如果不把SDT融合在Bison里，那还得根据生成好的树去看每一步推导用了啥产生式，进而确定具体的动作。

https://jielahou.com/compile/lab2.html

**Comment by yinzi001** [comment]
非常感谢，
